rotkehlchen/api/v1/schemas.py

```diff
@@ -704,7 +704,10 @@ def make_history_event_filter(
         }
 
         filter_query: HistoryEventFilterQuery | (EvmEventFilterQuery | EthStakingEventFilterQuery)
-        if should_query_evm_event or location_labels is not None:  # use evm event filter since only evm events have the "address" column for counterparty checks  # noqa: E501
+        if should_query_evm_event or (
+            location_labels is not None and
+            all(is_checksum_address(address) for address in location_labels)
```

Comment:
Don't we have something like `is_evm_address`?

Also would it not make sense here to change the logic more? `should_query_evm_event` should not be True here. This seems like the core of the issue. You should not get here in the first place.

Why would